### PR TITLE
Itc web worker

### PR DESCRIPTION
**Pull request by cquiroz**
This is a largish PR that moves requesting ITC data via the web worker. This doesn't really add new functionality but it would let us add much better local caching in a future PR.

A big change is a new intermediate project containing code to communicate with the graphql backend. This lets either the web worker or explore do backend queries

### CODE REVIEW

File: common-queries/src/main/scala/queries/schemas/odb/ObsQueries.scala [hunk 1]
```diff
@@ -40,6 +37,10 @@ import java.time.Instant
 import scala.collection.immutable.SortedMap
 
 object ObsQueries {
+  implicit class ListOps[A](val list: List[A]) {
```
Comment (toddburnside) [inline common-queries/src/main/scala/queries/schemas/odb/ObsQueries.scala:40]: extension method? 
We have this someplace common don't we? I assume it isn't visible from here?

File: explore/src/main/scala/explore/config/SpectroscopyModesTable.scala [hunk 2]
```diff
@@ -138,37 +149,33 @@ object SpectroscopyModesTable {
   val formatSlitLength: ModeSlitSize => String = ss =>
     f"${ModeSlitSize.milliarcseconds.get(ss.size).setScale(0, BigDecimal.RoundingMode.DOWN)}%1.0f"
 
-  def formatGrating(grating: InstrumentRow#Grating): String = grating match {
+  def formatGrating(grating: InstrumentRow#Grating): String = grating match
```
Comment (toddburnside) [inline explore/src/main/scala/explore/config/SpectroscopyModesTable.scala:152]: Are we dropping braces in some places?

File: explore/src/main/scala/explore/config/SpectroscopyModesTable.scala [hunk 3]
```diff
@@ -385,26 +389,48 @@ object SpectroscopyModesTable {
...
-      .useSerialStateView(
-        // .useState(
+      .useState {
         ItcResultsCache(
-          Map.empty[ITCRequestParams, EitherNec[ItcQueryProblems, ItcResult]]
+          Map.empty[ItcRequestParams, EitherNec[ItcQueryProblems, ItcResult]]
         )
+      }
+      // itcProgress
+      .useState(none[Progress])
```
Comment (toddburnside) [inline explore/src/main/scala/explore/config/SpectroscopyModesTable.scala:400]: I'm trying to understand how this works with a `worker` now. Before, with a single effect, this component would only get ITC results that it requested, right? What happens now if you submit a bunch of requests on one observation and switch to a new observation which submits a bunch of new requests? Will the new "observation" be processing responses to the requests from the old observation? Or does something prevent that?

File: common-queries/src/main/scala/queries/schemas/odb/ObsQueries.scala (hunk 1 above)
Comment (cquiroz) [reply common-queries/src/main/scala/queries/schemas/odb/ObsQueries.scala:40]: We do but it maybe hard to have it in common so I copied it

File: explore/src/main/scala/explore/config/SpectroscopyModesTable.scala (hunk 2 above)
Comment (cquiroz) [reply explore/src/main/scala/explore/config/SpectroscopyModesTable.scala:152]: You can drop them in some places yes

File: explore/src/main/scala/explore/config/SpectroscopyModesTable.scala (hunk 3 above)
Comment (cquiroz) [reply explore/src/main/scala/explore/config/SpectroscopyModesTable.scala:400]: Good question 🤔

Comment (toddburnside) [reply explore/src/main/scala/explore/config/SpectroscopyModesTable.scala:400]: Seems like that would throw off the Progress if nothing else.

Comment (cquiroz) [reply explore/src/main/scala/explore/config/SpectroscopyModesTable.scala:400]: Well progress gets reset if conditions change but I haven't thought about the case where we switch to another page
Maybe we need to index the cache by obsid

File: common-queries/src/main/scala/queries/schemas/odb/ObsQueries.scala [hunk 4]
```diff
@@ -40,6 +37,10 @@ import java.time.Instant
...
+    def toSortedMap[K: Ordering, V](getKey: A => K, getValue: A => V = identity[A](_)) =
```
Comment (rpiaggio) [inline common-queries/src/main/scala/queries/schemas/odb/ObsQueries.scala:41]: Why can't we use the one here https://github.com/gemini-hlsw/explore/blob/master/common/src/main/scala/explore/implicits.scala#L64 ?

Comment (toddburnside) [reply common-queries/src/main/scala/queries/schemas/odb/ObsQueries.scala:41]: I think right now the `common` module depends on this new module.

Comment (cquiroz) [reply common-queries/src/main/scala/queries/schemas/odb/ObsQueries.scala]: I moved it now to `model`

**Review by toddburnside — APPROVED** [review]
LGTM

**Pull request merged**